app/javascript/utilities/dragAndUpload.js

```diff
@@ -0,0 +1,12 @@
+import { generateMainImage } from '../article-form/actions';
+import { validateFileInputs } from '../packs/validateFileInputs';
+
+export function dragDrop(e, handleImageSuccess, handleImageFailure) {
```

Comment:
Although a small function, some tests can be added. It's easy to mock the Event object and the `handleImageSuccess` and `handleImageFailure` arguments can be easily mocked via `jest.fn()`.